var/spack/repos/builtin/packages/openfoam-org/50-etc.patch

```diff
@@ -0,0 +1,24 @@
+--- OpenFOAM-5.x.org/etc/bashrc	2017-07-25 18:43:40.000000000 +0200
++++ OpenFOAM-5.x/etc/bashrc	2017-12-11 13:36:09.479818186 +0100
+@@ -42,15 +42,17 @@
+ #
+ # Please set to the appropriate path if the default is not correct.
+ #
+-[ ${BASH_SOURCE:-$0} ] && \
+-export FOAM_INST_DIR=$(cd $(dirname ${BASH_SOURCE:-$0})/../.. && pwd -P) || \
```

Comment:
Change to handle zsh input as well.

Reply:
Using `export VAR=...` followed by a logical `||` does not make any sense to me at all. Quoting from bash(1) -

    export returns an exit status of 0 unless an invalid option is encountered,
    one of the names is not a valid shell variable name,
    or -f is supplied with a name that is not a function.


So in other words, the properly formatted `export` never fails for the
unpatched version. If it is desired to check the success/failure of
the `$(cd ...)` statement, then the patched version is the correct way
to do it.